test/awss3.jl

```diff
@@ -156,7 +156,7 @@ function awss3_tests(config)
         # https://github.com/samoconnor/AWSS3.jl/issues/24
         ctype(key) = s3_get_meta(bucket_name, key)["Content-Type"]
 
-        for k in ["file.foo", "file", "file_html", "file/html", "foobar.html/file.htm"]
+        for k in ["file.foo", "file", "file_html", "file.d/html", "foobar.html/file.htm"]
```

Comment:
what is this change for?

Reply:
Because `file` already exists and is non-empty, an error is thrown when writing to `file/html`, at least in min.io.  I'm not sure how this was ever allowed, it makes sense for it to be an error.  I'm not 100% sure that it is also an error in AWS but it really should be.

Note that if `file` is *empty* (0-bytes) then creating `file/html` *is* allowed.  This seems to be one of the accommodations that was made to emulate empty directories.

Reply:
ah ok. That makes sense. I see in the next few lines we do
```julia
            is_aws(config) && k == "file" && continue
```
so I guess we are skipping that test on AWS, which makes sense since it sounds like it should fail-- looks like I introduced that check in #163. Before that, it was https://github.com/JuliaCloud/AWSS3.jl/blame/8ace53a78109d040d8686d5c67f296396f1fec11/test/awss3.jl#L142-L151 -- not sure why that workd? But it was in the tests for years and I don't think they were failing.

Reply:
Yeah, I really don't know, the situation is very confusing across multiple versions of both minio and AWS, and I'm rather doubtful all the changes made to AWS are even documented.  Regardless: it was definitely failing in some cases, it doesn't seem as if it should succeed, doesn't seem appropriate to test it.